Starlight is incompatible with the Astro build.format preserve option

### What version of `starlight` are you using?

0.21.5

### What version of `astro` are you using?

4.6.3

### What package manager are you using?

pnpm

### What operating system are you using?

Mac

### What browser are you using?

Chrome

### Describe the Bug

when I add the options 
```
    trailingSlash: 'always',
    build: {
      format: 'preserve',
    },
```
to an existing otherwise working starlight project, it stops working with the error
```
13:16:42 [ERROR] Cannot read properties of undefined (reading 'addBase')
  Stack trace:
    at formatPath (/home/projects/withastro-starlight-7ibewh/node_modules/@astrojs/starlight/utils/createPathFormatter.ts:25:25)
    [...] See full stack trace in the browser, or rerun with --verbose.
```


### Link to Minimal Reproducible Example

https://stackblitz.com/edit/withastro-starlight-7ibewh?file=astro.config.mjs

### Participation

- [ ] I am willing to submit a pull request for this issue.

--- a/.changeset/curvy-pots-ring.md
+++ b/.changeset/curvy-pots-ring.md
@@ -1,0 +1,5 @@
+---
+'@astrojs/starlight': patch
+---
+
+Fixes support for Astro’s `build: { format: 'preserve' }` configuration option

--- a/packages/starlight/utils/createPathFormatter.ts
+++ b/packages/starlight/utils/createPathFormatter.ts
@@ -12,15 +12,18 @@ interface FormatPathOptions {
 	trailingSlash?: AstroConfig['trailingSlash'];
 }
 
+const defaultFormatStrategy = {
+	addBase: pathWithBase,
+	handleExtension: (href: string) => stripHtmlExtension(href),
+};
+
 const formatStrategies = {
 	file: {
 		addBase: fileWithBase,
 		handleExtension: (href: string) => ensureHtmlExtension(href),
 	},
-	directory: {
-		addBase: pathWithBase,
-		handleExtension: (href: string) => stripHtmlExtension(href),
-	},
+	directory: defaultFormatStrategy,
+	preserve: defaultFormatStrategy,
 };
 
 const trailingSlashStrategies = {
@@ -34,7 +37,6 @@ function formatPath(
 	href: string,
 	{ format = 'directory', trailingSlash = 'ignore' }: FormatPathOptions
 ) {
-	// @ts-expect-error — TODO: add support for `preserve` (https://github.com/withastro/starlight/issues/1781)
 	const formatStrategy = formatStrategies[format];
 	const trailingSlashStrategy = trailingSlashStrategies[trailingSlash];
 

